### PR TITLE
Generic localization between odf and odfmco

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Signed-off-by: Gowtham Shanmugasundaram <gshanmug@redhat.com>

### CODE REVIEW

File: locales/en/plugin__odf-console.json [hunk 1]
```diff
@@ -28,6 +28,15 @@
   "DRPolicies": "DRPolicies",
   "Asynchronous": "Asynchronous",
   "Synchronous": "Synchronous",
+  "StorageClasses": "StorageClasses",
+  "Replicas": "Replicas",
+  "UsedCapacity": "UsedCapacity",
+  "Mirroring Status": "Mirroring Status",
+  "Overall Image Health": "Overall Image Health",
+  "Compression Status": "Compression Status",
+  "Compression Savings": "Compression Savings",
```
[the remark this answers is not in the document]
Comment (SanjalKatiyar) [reply locales/en/plugin__odf-console.json:37]: okay, this PR is not specifically adding these texts...

File: packages/shared/status/statuses.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as React from 'react';
-import { useTranslation } from 'react-i18next';
 import { InProgressIcon } from '@patternfly/react-icons';
+import { useSharedTranslation } from '../useTranslationHook';
```
Comment (SanjalKatiyar) [inline packages/shared/status/statuses.tsx:3]: nits: useTranslation is already provided by i18n... `useCustomTranslationHook` or `useSharedTranslationHook` as a name for the file would be better IMO.
```suggestion
import { useSharedTranslation } from '../useSharedTranslationHook';
```

File: locales/en/plugin__odf-console.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply locales/en/plugin__odf-console.json:37]: Yep, looks like this was missed in already merged PR. i will change this.

File: packages/shared/status/statuses.tsx (hunk 2 above)
Comment (GowthamShanmugam) [reply packages/shared/status/statuses.tsx:3]: ack

**Comment by GowthamShanmugam** [comment]
@SanjalKatiyar @bipuladh Changes done

File: packages/shared/useCustomTranslationHook.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { useTranslation } from 'react-i18next';
+
+
+export const NS = process.env.NS;
+
+export const useCustomTranslation = () => {
+    return useTranslation(NS);
+};
```
Comment (afreen23) [inline packages/shared/useCustomTranslationHook.ts:8]:
```suggestion
export const useCustomTranslation = () => useTranslation(NS);
```

File: packages/shared/useCustomTranslationHook.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { useTranslation } from 'react-i18next';
+
+
+export const NS = process.env.NS;
```
Comment (afreen23) [inline packages/shared/useCustomTranslationHook.ts:4]:
```suggestion
export const NS = process.env.I8N_NS;
```

Comment (afreen23) [reply packages/shared/useCustomTranslationHook.ts:4]: It will give more clarity about NS

File: webpack.config.ts [hunk 5]
```diff
@@ -5,6 +5,11 @@ import { ConsoleRemotePlugin } from '@openshift-console/dynamic-plugin-sdk-webpa
 import * as CopyWebpackPlugin from 'copy-webpack-plugin';
 import * as webpack from 'webpack';
 
+const resolveLocale = (dirName: string, ns: string) =>
+  ['en', 'ja', 'ko', 'zh'].reduce((acc, lang) => ([...acc, { from: path.resolve(dirName, `locales/${lang}/plugin__*.json`), to: `locales/${lang}/${ns}.[ext]`}]), [])
```
Comment (afreen23) [inline webpack.config.ts:9]: Create a constant for the languages used:
export const LANGUAGES =  ['en', 'ja', 'ko', 'zh'];
```suggestion
 LANGUAGES.reduce((acc, lang) => ([...acc, { from: path.resolve(dirName, `locales/${lang}/plugin__*.json`), to: `locales/${lang}/${ns}.[ext]`}]), [])
```

**Review by afreen23 — CHANGES_REQUESTED** [review]

**Pull request merged**